pkg/http/sse_test.go

```diff
@@ -162,6 +163,42 @@ func TestSSE(t *testing.T) {
 		response = <-responses
 		require.Nil(t, response, "should receive no update")
 	})
+
+	t.Run("event_stream_updated_after_context_closed", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		responses := newSSERequest(t, ctx, server.URL+"/posts/"+delayedID)
+
+		response, ok := <-responses
+		require.True(t, ok)
+		post := Post{}
+		err = json.Unmarshal(response, &post)
+		require.NoError(t, err)
+		require.Equal(t, delayedID, post.ID)
+
+		payload, err := json.Marshal(PostUpdated{ID: delayedID})
+		require.NoError(t, err)
+
+		msg := message.NewMessage(watermill.NewUUID(), payload)
+		err = pubsub.Publish(postUpdatedTopic, msg)
+		require.NoError(t, err)
+
+		// Wait until ByID() starts processing
+		time.Sleep(time.Millisecond * 100)
+
+		cancel()
+
+		// Wait until ByID() finishes
+		time.Sleep(time.Millisecond * 150)
+
+		select {
+		case response, ok = <-responses:
+			require.False(t, ok)
+			require.Nilf(t, response, "should receive no update, got %s", string(response))
```

Comment:
not sure if `Nilf` will work (guess it will, you tested it). Anyway, `Emptyf` will probably be more readable